src/components/VideoPlayer/VideoPlayerControls/VolumeButton/index.tsx

```diff
@@ -73,6 +73,13 @@ function VolumeButton({style, small = false}: VolumeButtonProps) {
         runOnJS(updateIcon)(volume.get());
     }, [volume]);
 
+    const toggleMute = () => {
+        if (volume.get() !== 0) {
+            lastNonZeroVolume.set(volume.get());
+        }
+        updateVolume(volume.get() === 0 ? lastNonZeroVolume.get() : 0);
+    };
```

Comment:
How about moving this logic to the Volume context? Additionally, I strongly recommend adding a comment to clearly explain its purpose.